jenkins/automation/splint-memory-leak.groovy

```diff
@@ -0,0 +1,93 @@
+                }
+            }
+        }
+
+        stage ('Execute Memory Analysis') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: '', returnStatus: true, script: '''
+                yum install splint -y
+                mkdir -p artifacts
+                for file in  $(find $PWD -type f -regextype posix-egrep -regex ".+\\\\.(c|h|lcl)$"); do splint +trytorecover -preproc -warnposix $file >> artifacts/splint-analysis.log ; done || echo "Log Generated"
+                '''
+            }
+        }
+    }
+
+    post {
+
+        cleanup {
+                script {
+                    // Archive Deployment artifacts in jenkins build
+                    archiveArtifacts artifacts: "artifacts/*.*", onlyIfSuccessful: false, allowEmptyArchive: true
+                }
+        }
+
+        always {
+            script {
+                // Jenkins Summary
+                clusterStatus = ""
+                if ( currentBuild.currentResult == "SUCCESS" ) {
+                    MESSAGE = "Memory Leak Analysis Execution Success for the build ${build_id}"
+                    ICON = "accept.gif"
+                    STATUS = "SUCCESS"
+                } else if ( currentBuild.currentResult == "FAILURE" ) {
+                    manager.buildFailure()
+                    MESSAGE = "Memory Leak Analysis Execution Failed for the build ${build_id}"
+                    ICON = "error.gif"
+                    STATUS = "FAILURE"
+
+                } else {
+                    manager.buildUnstable()
+                    MESSAGE = "Memory Leak Analysis Execution is Unstable for the build ${build_id}"
+                    ICON = "warning.gif"
+                    STATUS = "UNSTABLE"
+                }
+
+                MemoryLeakAnalysisStatusHTML = "<pre>${clusterStatus}</pre>"
+
+                manager.createSummary("${ICON}").appendText("<h3>Memory Leak Analysis Execution ${currentBuild.currentResult} </h3><p>Please check <a href=\"${BUILD_URL}/artifact/artifacts/splint-analysis.log\">Memory Leak Analysis logs</a> for Memory leak errors</p><p>Please check <a href=\"${BUILD_URL}/console\">Memory Leak Analysis Execution logs</a> for more info</p>", false, false, false, "red")
+
+                // Email Notification
+                env.build_stage = "${build_stage}"
+                env.cluster_status = "${MemoryLeakAnalysisStatusHTML}"
+                def recipientProvidersClass = [[$class: 'RequesterRecipientProvider']]
+                mailRecipients = "CORTX.DevOps.RE@seagate.com"
+                emailext (
+                    body: '''${SCRIPT, template="devops-automation-email.template"}''',
+                    mimeType: 'text/html',
+                    subject: "[Jenkins Build ${currentBuild.currentResult}] : ${env.JOB_NAME}",
```

Comment:
can you add `MESSAGE` as a subject?

Reply:
What `MESSAGE` need to be added? Can you elaborate more?

Reply:
Fixed. Please review again.